cadcdata/cadcdata/tests/test_inventory.py

```diff
@@ -0,0 +1,571 @@
+        assert expected_content == f.read()
+    os.remove(file_name)
+    # do it again with the file now open
+    response = Mock()
+    response.headers = {
+        'content-disposition': 'inline; filename=orig_file_name',
+        'digest': 'md5={}'.format(b64encoded)}
+    response.raw.read.side_effect = file_chunks
+    basews_mock.return_value.get.return_value = response
+    with open(file_name, 'wb') as f:
+        client.cadcget('cadc:TEST/afile', dest=f)
+    with open(file_name, 'rb') as f:
+        assert expected_content == f.read()
+    os.remove(file_name)
+
+    # repeat test with a bad md5
+    file_name = 'bfile.txt'
+    file_content = 'ABCDEFGH12345'
+    file_chunks = [file_content.encode(), ''.encode()]
+    decoded_file_content = 'MNOPRST6789'
+    decoded_file_chunks = [decoded_file_content.encode(), ''.encode()]
+    response = Mock()
+    wrong_b64encoded = base64.b64encode('abc'.encode('ascii')).decode('ascii')
+    response.headers = {
+        'digest': 'md5={}'.format(wrong_b64encoded),
+        'content-disposition': 'inline; filename={}'.format(file_name)}
+    response.raw.read.side_effect = file_chunks
+    response.raw._decode.side_effect = decoded_file_chunks
+    basews_mock.return_value.get.return_value = response
+    client = StorageInventoryClient(auth.Subject())
+    with pytest.raises(exceptions.HttpException):
+        client.cadcget('cadc:TEST:bfile.txt', file_name)
+
+    # test process_bytes and send the content to /dev/null after.
+    def concatenate_chunks(chunk):
+        global mycontent
+        mycontent = '{}{}'.format(mycontent, chunk.decode())
+
+    file_name = 'bfile.txt'
+    file_content = 'ABCDEFGH12345'
+    file_chunks = [file_content[i:i + 5].encode()
+                   for i in xrange(0, len(file_content), 5)]
+    file_chunks.append('')  # last chunk is empty
+    response = Mock()
+    response.headers = {
+        'content-disposition': 'inline; filename={}.gz'.format(file_name)}
+    response.raw.read.side_effect = file_chunks
+    basews_mock.return_value.get.return_value = response
+    client = StorageInventoryClient(auth.Subject())
+    # md5_check does not take place because no content-MD5 received
+    # from server
+    client.cadcget('cadc:TEST/afile', dest='/dev/null',
+                   process_bytes=concatenate_chunks)
+    assert file_content == mycontent
+
+    # failed md5 checksum
+    response = Mock()
+    response.headers = {
+        'content-disposition': 'inline; filename={}.gz'.format(file_name),
+        'digest': 'md5={}'.format(
+            base64.b64encode('f00'.encode('ascii')).decode('ascii'))}
+    response.raw.read.side_effect = file_chunks
+    basews_mock.return_value.get.return_value = response
+    client = StorageInventoryClient(auth.Subject())
+    with pytest.raises(exceptions.HttpException):
+        client.cadcget('cadc:TEST/afile', dest='/dev/null',
+                       process_bytes=concatenate_chunks)
+
```

Comment:
Possible test case - what happens to local naming when id = cadc:CFHT/12345670.fits.fz, since there is automatic decompression by the library?

Reply:
No automatic decompression supported yet.